sympy/functions/combinatorial/factorials.py

```diff
@@ -1075,14 +1075,11 @@ def _eval_rewrite_as_Piecewise(self, n, k, **kwargs):
                     Or(And(n*k > 0, k - n > 0),
                         And(n > 0, k < 0))))
             ),
-            (Product((n - k + i)/i,(i, 1,
-                Piecewise(
-                (Piecewise(
-                (-k + n, (k < n) & (n < 2*k)),
-                (k, True)),
-                And(n > 0, k > 0)),
-                (n - k, And(n >= 0, k <  0)),
-                (k, True)))), isint(k)),
+            (Piecewise(
+                    (Product((k + i)/i, (i, 1, n - k)), And(n < 0, k < 0)),
```

Comment:
I like how this looks. Above we handled the case of `n*k >0 and k>n` so the first case is ok and we also took care of neg k and positive n above so all that we have here is negative n and positive k for the second case...I think.

Reply:
I also had it equated with the way binomial behaved, I didn't get any assertion errors 😄 
shall I add a `.equals` test for that too ?

Reply:
> with the way binomial behaved

I thought I roughly did, too. But it's been a moving target. And perhaps my memory is a little rough, too ;-) 

What sort of `equals` tests are you thinking of?

Reply:
Here's an example
```python3
>>> binomial(x, y).equals(binomial(x,y).rewrite(Piecewise))
False
>>> x, y = symbols('x y', integer = True)
>>> binomial(x, y).equals(binomial(x,y).rewrite(Piecewise))
>>> 
```
They fail currently tho 😅